sympy/functions/special/bessel.py

```diff
@@ -216,11 +218,38 @@ def _eval_rewrite_as_bessely(self, nu, z, **kwargs):
     def _eval_rewrite_as_jn(self, nu, z, **kwargs):
         return sqrt(2*z/pi)*jn(nu - S.Half, self.argument)
 
+    def _eval_as_leading_term(self, x, cdir=0):
+        nu, z = self.args
+        arg = z.as_leading_term(x)
+        if x in arg.free_symbols:
+            return z**nu
+        else:
+            return self.func(*self.args)
+
     def _eval_is_extended_real(self):
         nu, z = self.args
         if nu.is_integer and z.is_extended_real:
             return True
 
+    def _eval_nseries(self, x, n, logx, cdir=0):
+        from sympy.series.order import Order
+        nu, z = self.args
+
+        # In case of powers less than 1, number of terms need to be computed
+        # separately to avoid repeated callings of _eval_nseries with wrong n
+        try:
+            _, exp = z.leadterm(x)
+            newn = ceiling(n/exp) if not exp is S.Zero else n
```

Comment:
`exp` could also be negative. I would leave the above `leadterm` line alone in the try block and replace the condition `z.limit(x, 0) is S.Zero` by `exp > 0` (and only then compute `newn`).